secrets/store_test.go

```diff
@@ -77,6 +137,29 @@ func TestGetSimpleSecret(t *testing.T) {
 				}
 				defer store.Close()
 
+				secret, err := store.GetSimpleSecret(tt.key)
+				if tt.expectedError == nil && err != nil {
+					t.Fatal(err)
+				}
+				if tt.expectedError != nil && err.Error() != tt.expectedError.Error() {
+					t.Fatalf("expected error %v, actual: %v", tt.expectedError, err)
+				}
+				if !reflect.DeepEqual(secret, tt.expected) {
+					t.Fatalf("expected %+v, actual: %+v", tt.expected, secret)
+				}
+			},
+		)
+	}
+	for _, tt := range tests {
+		t.Run(
+			tt.name,
```

Comment:
this will run the subtests with the same name as from line 132. which will make the output very confusing.

either construct the name to add `-file`/`-dir` suffix (e.g. make line 132 `tt.name + "-file"` and here `tt.name + "-dir"`, or run another level of subtests with names `file`/`dir`.

I would strongly prefer to add another level of subtests. you can also make it another level of for loop to reuse the majority of the code:

```go
	for _, tt := range tests {
		t.Run(
			tt.name,
			func(t *testing.T) {
				for label, path := range map[string]string{
					"file": tmpPath,
					"dir": dirCSI,
				} {
					t.Run(label, func(t *testing.T) {
						store, err := secrets.NewStore(context.Background(), path, log.TestWrapper(t))
						if err != nil {
							t.Fatal(err)
						}
						t.Cleanup(func() { store.Close() })

						secret, err := store.GetSimpleSecret(tt.key)
						if tt.expectedError == nil && err != nil {
							t.Fatal(err)
						}
						if tt.expectedError != nil && err.Error() != tt.expectedError.Error() {
							t.Fatalf("expected error %v, actual: %v", tt.expectedError, err)
						}
						if !reflect.DeepEqual(secret, tt.expected) {
							t.Fatalf("expected %+v, actual: %+v", tt.expected, secret)
						}
					})
				}
			},
		)
	}
```

Reply:
you are correct